### PR TITLE
*: add a config to enable write statements to read on TiFlash

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: gengliqi <gengliqiii@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/37515

Problem Summary:

### What is changed and how it works?
Add a session config `tidb_enable_tiflash_read_for_write_stmt`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by LittleFall — APPROVED** [review]
agree with https://github.com/pingcap/tidb/pull/37516/files#r959628864, others LGTM

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -234,6 +234,9 @@ const (
 
 	// TiFlashFastScan indicates whether use fast scan in tiflash .
 	TiFlashFastScan = "tiflash_fastscan"
+
+	// TiFlashReadForWriteStmt indicates whether to enable write stmts to read on TiFlash.
+	TiFlashReadForWriteStmt = "tiflash_read_for_write_stmt"
```
[the remark this answers is not in the document]
Comment (gengliqi) [reply sessionctx/variable/tidb_vars.go:239]: `select` is considered to be part of DML(strictly speaking, it's DQL). What I want to express is that the read part in `insert/update/delete` SQL can be pushed down to TiFlash. 
How about `tidb_enable_tiflash_for_all_dml` or `tidb_enable_tiflash_for_write_stmt`?

File: sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -1791,6 +1791,10 @@ var defaultSysVars = []*SysVar{
 			}
 			return nil
 		}},
+	{Scope: ScopeSession, Name: TiDBEnableTiFlashReadForWriteStmt, Value: BoolToOnOff(DefTiDBEnableTiFlashReadForWriteStmt), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```
Comment (XuHuaiyu) [inline sessionctx/variable/sysvar.go:1794]: ScopeGlobal is not needed?

Comment (gengliqi) [reply sessionctx/variable/sysvar.go:1794]: ScopeSession is enough for testing at present.

**Review by XuHuaiyu — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**